internal/session/handle_login.go

```diff
@@ -24,9 +31,21 @@ func (s *Session) handleLogin(ctx context.Context, tag string, cmd *proto.Login,
 
 	state, err := s.backend.GetState(cmd.GetUsername(), cmd.GetPassword(), s.sessionID)
 	if err != nil {
+		s.loginErrorCount++
```

Comment:
Wouldn't it be better to have this check on the the user account rather than the session? An evil user could just launch N connection in parallel. If you have this check on the user, you can scale the login timeout based on the number of attempts. 

One should still disconnect session after 3 tries, but all login attempts should have the timeout scaling applied.

Reply:
we can't do it on user account  side since we also need to track bad login attempts with unknown account/user

Reply:
I did modification to handle the timeout at a more global level (backend) rather that only on sessions.